nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/status/history/ConnectionStatusDescriptor.java

```diff
@@ -79,4 +95,52 @@ public String getField() {
     public MetricDescriptor<ConnectionStatus> getDescriptor() {
         return descriptor;
     }
+
+    private static final Logger LOG = LoggerFactory.getLogger(ConnectionStatusDescriptor.class);
+
+    private static NiFiProperties nifiProperties;
+
+    public static void setNifiProperties(NiFiProperties nifiProperties) {
+        ConnectionStatusDescriptor.nifiProperties = nifiProperties;
+    }
+
```

Comment:
This could be my ignorance, but are labels intended to be immutable? Is modifying them a bad thing?

Reply:
It was labeled as 'final' originally, but I'm not sure if that is because it is thought that labels are always immutable or rather the case for these particular labels changing was not foreseen. I wanted a means for the user to know when viewing the graph that the reason it was at time 0 was not due to a queue overflow but rather that the analytic engine was not being utilized. I'm open to other means of getting that info across if you have any suggestions.

Reply:
It might be good to get @markap14 's feedback here since he's the original author.

Reply:
Hi @markap14, would you be able to take a look at this and offer your thoughts? Thanks.

Reply:
@m-hogue I haven't received any feedback from @markap14 at this point. Should we continue to wait or consider merging?